resource/user/preparer.go

```diff
@@ -72,6 +73,12 @@ type Preparer struct {
 	// HomeDir must also be indicated if MoveDir is set to true.
 	MoveDir bool `hcl:"move_dir"`
 
+	// Expiry is the date on which the user account will be disabled. The date is
+	// specified in the format YYYY-MM-DD. If not specified, the default expiry
+	// date specified by the EXPIRE variable in /ect/default/useradd, or an empty
+	// string (no expiry) will be used by default.
+	Expiry time.Time `hcl:"expiry"`
```

Comment:
Should this be `*time.Time`?  Otherwise any other changes to a user will have the side effect of setting an expiry since we'll always be setting it.